src/DetailsView/reports/report-generator.ts

```diff
@@ -18,24 +18,22 @@ export class ReportGenerator {
         private assessmentReportHtmlGenerator: AssessmentReportHtmlGenerator,
     ) {}
 
-    public generateName(scanDate: Date, pageTitle: string): string {
-        return this.reportNameGenerator.generateName(scanDate, pageTitle);
+    public generateName(baseName: string, scanDate: Date, pageTitle: string): string {
+        return this.reportNameGenerator.generateName(baseName, scanDate, pageTitle);
     }
 
     public generateHtml(scanResult: ScanResults, scanDate: Date, pageTitle: string, pageUrl: string, description: string): string {
         return this.reportHtmlGenerator.generateHtml(scanResult, scanDate, pageTitle, pageUrl, description);
     }
 
     public generateAssessmentHtml(
-        deps: ReportGeneratorDeps,
         assessmentStoreData: IAssessmentStoreData,
         assessmentsProvider: IAssessmentsProvider,
         featureFlagStoreData: FeatureFlagStoreData,
         tabStoreData: ITabStoreData,
         description: string,
     ): string {
         return this.assessmentReportHtmlGenerator.generateHtml(
```

Comment:
In a more consistent world, we'd pass the `scanDate` to `assessmentReportHtmlGenerator.generateHtml` to ensure that the same date is used in both places (and from there to `assessmentReportModelBuilderFactory.create`) - but given that we're now only using the date (in local time zone) in the file name, there's not much of a chance of a user-visible mismatch. Given that this is an existing inconsistency between the assessment and the automated checks report code, not a blocker.

Reply:
Agreed. Discussed offline with @peterdur and we will save this for the future.